main/rfc1867.c

```diff
@@ -1149,9 +1149,39 @@ SAPI_API SAPI_POST_HANDLER_FUNC(rfc1867_post_handler) /* {{{ */
 				snprintf(lbuf, llen, "%s[name]", param);
 			}
 			register_http_post_files_variable(lbuf, s, &PG(http_globals)[TRACK_VARS_FILES], 0);
-			efree(filename);
 			s = NULL;
 
+			/* Add $foo_fullpath */
+			if (llen < strlen(param) + MAX_SIZE_OF_INDEX + 1) {
+				llen = (int)strlen(param);
+				lbuf = (char *) safe_erealloc(lbuf, llen, 1, MAX_SIZE_OF_INDEX + 1);
+				llen += MAX_SIZE_OF_INDEX + 1;
+			}
+
+			if (is_arr_upload) {
+				if (abuf) efree(abuf);
+				abuf = estrndup(param, strlen(param)-array_len);
+				snprintf(lbuf, llen, "%s_fullpath[%s]", abuf, array_index);
+			} else {
+				snprintf(lbuf, llen, "%s_fullpath", param);
+			}
+
+			/* Add full path of supplied file for folder uploads via 
+			 * <input type="file" name="files" multiple webkitdirectory>
+			 */
+			if (!is_anonymous) {
+				safe_php_register_variable(lbuf, filename, strlen(filename), NULL, 0);
+			}
```

Comment:
This is actually dead code. I've removed it in https://github.com/php/php-src/commit/61ba79df17867715b265de8b47238f96e62edf9c and you can drop it here as well (together with the snprintf above).